com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.Debug.cs

```diff
@@ -6,15 +6,197 @@ namespace UnityEngine.Rendering.HighDefinition
+        public DebugDisplaySettings debugDisplaySettings { get { return m_DebugDisplaySettings; } }
+        static DebugDisplaySettings s_NeutralDebugDisplaySettings = new DebugDisplaySettings();
+        internal DebugDisplaySettings m_CurrentDebugDisplaySettings;
+
+        void InitializeDebug()
+        {
+            m_DebugViewMaterialGBuffer = CoreUtils.CreateEngineMaterial(defaultResources.shaders.debugViewMaterialGBufferPS);
+            m_DebugViewMaterialGBufferShadowMask = CoreUtils.CreateEngineMaterial(defaultResources.shaders.debugViewMaterialGBufferPS);
+            m_DebugViewMaterialGBufferShadowMask.EnableKeyword("SHADOWS_SHADOWMASK");
+            m_DebugDisplayLatlong = CoreUtils.CreateEngineMaterial(defaultResources.shaders.debugDisplayLatlongPS);
+            m_DebugFullScreen = CoreUtils.CreateEngineMaterial(defaultResources.shaders.debugFullScreenPS);
+            m_DebugColorPicker = CoreUtils.CreateEngineMaterial(defaultResources.shaders.debugColorPickerPS);
+            m_DebugExposure = CoreUtils.CreateEngineMaterial(defaultResources.shaders.debugExposurePS);
+            m_DebugViewTilesMaterial = CoreUtils.CreateEngineMaterial(defaultResources.shaders.debugViewTilesPS);
+            m_DebugHDShadowMapMaterial = CoreUtils.CreateEngineMaterial(defaultResources.shaders.debugHDShadowMapPS);
+            m_DebugDensityVolumeMaterial = CoreUtils.CreateEngineMaterial(defaultResources.shaders.debugDensityVolumeAtlasPS);
+            m_DebugBlitMaterial = CoreUtils.CreateEngineMaterial(defaultResources.shaders.debugBlitQuad);
+#if ENABLE_VIRTUALTEXTURES
+            m_VTDebugBlit = CoreUtils.CreateEngineMaterial(defaultResources.shaders.debugViewVirtualTexturingBlit);
+#endif
+        }
+
+        void CleanupDebug()
+        {
+            CoreUtils.Destroy(m_DebugViewMaterialGBuffer);
+            CoreUtils.Destroy(m_DebugViewMaterialGBufferShadowMask);
+            CoreUtils.Destroy(m_DebugDisplayLatlong);
+            CoreUtils.Destroy(m_DebugFullScreen);
+            CoreUtils.Destroy(m_DebugColorPicker);
+            CoreUtils.Destroy(m_DebugExposure);
+            CoreUtils.Destroy(m_DebugViewTilesMaterial);
+            CoreUtils.Destroy(m_DebugHDShadowMapMaterial);
+            CoreUtils.Destroy(m_DebugDensityVolumeMaterial);
+            CoreUtils.Destroy(m_DebugBlitMaterial);
+#if ENABLE_VIRTUALTEXTURES
+            CoreUtils.Destroy(m_VTDebugBlit);
+#endif
+        }
+
+        internal bool showCascade
+        {
+            get => m_DebugDisplaySettings.GetDebugLightingMode() == DebugLightingMode.VisualizeCascade;
+            set
+            {
+                if (value)
+                    m_DebugDisplaySettings.SetDebugLightingMode(DebugLightingMode.VisualizeCascade);
+                else
+                    m_DebugDisplaySettings.SetDebugLightingMode(DebugLightingMode.None);
+            }
+        }
+
+        static bool NeedColorPickerDebug(DebugDisplaySettings debugSettings)
```

Comment:
Why is this static and the other not? Is it called from outside and not visible in this PR?

Reply:
No reason in particular I guess. It was just copy pasted as is.
A case could be made that they should all be static since they are kind of "pure" functions anyway